python/cudf/cudf/core/column/column.py

```diff
@@ -1939,20 +1939,18 @@ def as_column(
     * pyarrow array
     * pandas.Categorical objects
     """
-    if isinstance(arbitrary, ColumnBase):
-        if dtype is not None:
-            return arbitrary.astype(dtype)
+    if isinstance(arbitrary, (ColumnBase, cudf.Series, cudf.BaseIndex)):
+        if isinstance(arbitrary, cudf.Series):
+            column = arbitrary._column
+        elif isinstance(arbitrary, cudf.BaseIndex):
+            column = arbitrary._values
         else:
-            return arbitrary
-
-    elif isinstance(arbitrary, cudf.Series):
-        data = arbitrary._column
+            column = arbitrary
+        if column.dtype.kind == "f" and (nan_as_null is None or nan_as_null):
```

Comment:
Usually I tend to prefer some parameter handling at the start of the function, such as:
```python
if nan_as_null is None:
    nan_as_null = True
```